### PR TITLE
wikirate encoding issue

### PR DESCRIPTION
**Pull request by xithan**
this handles all these "invalid byte sequence in UTF-8" errors we are getting on wikirate

### CODE REVIEW

File: wagn/rails/controllers/card_controller.rb [hunk 1]
```diff
@@ -73,6 +73,11 @@ def per_request_setup
     Card::Cache.renew
     Card::Env.reset :controller=>self
     Card::Auth.set_current_from_session
+    
+    if params[:id] && !params[:id].valid_encoding?  # slightly better way to handle encoding issues (than the rescue in load_id)
+                                                    # we should find the place where we produce these bad urls
```
Comment (GerryG) [inline wagn/rails/controllers/card_controller.rb]: We produce them or they are external?  What do we know about where they come from?

Comment (ethn) [inline wagn/rails/controllers/card_controller.rb]: good question. we'll have to do some research on airbrake, I suppose.

Comment (xithan) [inline wagn/rails/controllers/card_controller.rb]: Unfortunately airbrake can't tell us where link is coming from. The last errors are all produced by +link_and_compare addresses, for example http://wikirate.org/Amalgamated_Plantations_Pvt_Ltd_+Human_Rights+L_Or%e9al+link_and_compare. The remote ip belongs Hetzner Online, that's Wikirate's server host, right?

Comment (GerryG) [inline wagn/rails/controllers/card_controller.rb]: I bet they are internal links.  Any ideas about where the third part of that name comes from in the code?  Is it a real cardname when you apply this patched code to it?  The percent encoding decoded should get you a UTF-8 codepoint, right?  UTF encodings are required for URLs, I would think.  I'm guessing it is actually the latin encoding in the patch.

**Comment by GerryG** [comment]
You need to catch it when it is generating the bad links.

In some way, it seems like a risky patch.  The URLs are probably incorrectly encoded.  Workaround isn't bad, but if true, will cause trouble with other sites (potentially).

File: wagn/rails/controllers/card_controller.rb (hunk 1 above)
Comment (xithan) [inline wagn/rails/controllers/card_controller.rb]: the problem are characters like é as in "L'oréal". they are part of card names but I can't find the place where the bad links are build. And it has nothing to do with recent changes. These errors occasionally occurred for months.

Comment (GerryG) [inline wagn/rails/controllers/card_controller.rb]: Could it be inside the smartname gem?

**Pull request merged**